gutenberg.php

```diff
@@ -181,12 +181,14 @@ function gutenberg_intercept_edit_post() {
 		return;
 	}
 
-	if ( isset( $_GET['post'] ) ) {
-		$post_ID = (int) $_GET['post'];
-		$post_id = $post_ID;
+	if ( empty( $_GET['post'] ) ) {
+		return;
 	}
 
-	if ( empty( $post_id ) ) {
+	$post_ID = (int) $_GET['post'];
+	$post_id = $post_ID;
+
+	if ( ! $post_id ) {
```

Comment:
When would this condition be triggered? Looks like dead code to me.

Reply:
Yeah, this is what @swissspidy and I were talking about. I only can think of incorrect usage of the functionality, for example, hand-crafting an invalid URL without a (valid) post ID.

Reply:
Empty value, negative number, zero, some string, ... All that would be `(int) 0`, and thus falsy.

Reply:
Ok, I understand, but when would `$post_id` be assigned zero, since it's indirectly assigned from `$_GET['post']` and we have the `empty( $_GET['post'] )` condition above to shortcut?

Reply:
When I put `?post=foo` in the URL, its not empty in terms of the first check ´empty()`, but after casting it to an integer, it is `0´, and thus _empty_.

Reply:
But yes, **personally**, I would have adapted the first check to include this already. This would have been a backward cincompatible change, though, as the current code sets the `$post_id` and `$post_ID` variables to that invalid value.

Should I adapt this?

Reply:
```
if ( empty( $_GET['post'] ) || ! is_numeric( $_GET['post'] ) ) {
	return;
}

$post_ID = (int) $_GET['post'];
$post_id = $post_ID;
```

Reply:
Yes, that seems improved to me, since if this conversation is any evidence, what was being accounted for was not entirely clear (though in retrospect makes sense, thanks for clarifying).

Re: Backwards incompatibility, I'd assume `post=foo` would break miserably in some other manner anyways, or at least not something we'd intend to support.

Reply:
Done.